src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -2,21 +2,21 @@
 
 public class IntLinkedList {
 
-	Node last;
-	int i;
+	private Node last;
+	private int i;
 
 	public void push(int i) {
-		if (last == null)
-			last = new Node(i);
+		if (getLast() == null)
+			setLast(new Node(i));
 		else {
-			last.next = new Node(i);
-			last.next.prev = last;
-			last = last.next;
+			getLast().setNext(new Node(i));
+			getLast().getNext().setPrev(getLast());
+			setLast(getLast().getNext());

```

Comment:
This is an exaggeration. Using private fields internally usually do not require encapsulation.